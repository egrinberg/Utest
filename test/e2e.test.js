const { run } = require('./run');

describe('end-to-end integration', () => {
  test('when no arguments passed prints error and help message', async () => {
    const { stderr, stdout, exitCode } = await run();
    expect(exitCode).toBe(1);
    expect(stderr).toMatchSnapshot();
    expect(stdout).toEqual('');
  });

  test('when argument --help is specified should print help message', async () => {
    const { stderr, stdout, exitCode } = await run('--help');
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual('');
  });

  test('when argument -h is specified should print help message', async () => {
    const { stderr, stdout, exitCode } = await run('-h');
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual('');
  });

  test('when argument -j without file is specified should print help message', async () => {
    const { stderr, stdout, exitCode } = await run('-j');
    expect(exitCode).toBe(1);
    expect(stdout).toEqual('');
    expect(stderr).toMatchSnapshot();
  });

  test('when argument -f without file is specified should print help message', async () => {
    const { stderr, stdout, exitCode } = await run('-f');
    expect(exitCode).toBe(1);
    expect(stdout).toEqual('');
    expect(stderr).toMatchSnapshot();
  });

  test('when argument --file without file is specified should print help message', async () => {
    const { stderr, stdout, exitCode } = await run('--file');
    expect(exitCode).toBe(1);
    expect(stdout).toEqual('');
    expect(stderr).toMatchSnapshot();
  });

  test('when argument -j -f without file is specified should print help message', async () => {
    const { stderr, stdout, exitCode } = await run('-j -f');
    expect(exitCode).toBe(1);
    expect(stdout).toEqual('');
    expect(stderr).toMatchSnapshot();
  });

  test('when argument --json -f without file is specified should print help message', async () => {
    const { stderr, stdout, exitCode } = await run('--json -f');
    expect(exitCode).toBe(1);
    expect(stdout).toEqual('');
    expect(stderr).toMatchSnapshot();
  });

  test('when argument --json --file without file is specified should print help message', async () => {
    const { stderr, stdout, exitCode } = await run('--json --file');
    expect(exitCode).toBe(1);
    expect(stdout).toEqual('');
    expect(stderr).toMatchSnapshot();
  });

  test('when argument -j --file without file is specified should print help message', async () => {
    const { stderr, stdout, exitCode } = await run('-j --file');
    expect(exitCode).toBe(1);
    expect(stdout).toEqual('');
    expect(stderr).toMatchSnapshot();
  });
});
