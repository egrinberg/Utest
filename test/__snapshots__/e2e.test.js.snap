// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`end-to-end integration when argument --help is specified should print help message 1`] = `
"Usage: Utest.js [option]

Options:
  -f, --file     Load a file                                          [required]
  -h, --help     Show help                                             [boolean]
  -v, --version  show version information                              [boolean]
  -V, --version  show version information                              [boolean]
  -j, --json     print in JSON format"
`;

exports[`end-to-end integration when argument -f without file is specified should print help message 1`] = `
"Usage: Utest.js [option]

Options:
  -f, --file     Load a file                                          [required]
  -h, --help     Show help                                             [boolean]
  -v, --version  show version information                              [boolean]
  -V, --version  show version information                              [boolean]
  -j, --json     print in JSON format

Not enough arguments following: f"
`;

exports[`end-to-end integration when argument --file without file is specified should print help message 1`] = `
"Usage: Utest.js [option]

Options:
  -f, --file     Load a file                                          [required]
  -h, --help     Show help                                             [boolean]
  -v, --version  show version information                              [boolean]
  -V, --version  show version information                              [boolean]
  -j, --json     print in JSON format

Not enough arguments following: file"
`;

exports[`end-to-end integration when argument -h is specified should print help message 1`] = `
"Usage: Utest.js [option]

Options:
  -f, --file     Load a file                                          [required]
  -h, --help     Show help                                             [boolean]
  -v, --version  show version information                              [boolean]
  -V, --version  show version information                              [boolean]
  -j, --json     print in JSON format"
`;

exports[`end-to-end integration when argument -j without file is specified should print help message 1`] = `
"Usage: Utest.js [option]

Options:
  -f, --file     Load a file                                          [required]
  -h, --help     Show help                                             [boolean]
  -v, --version  show version information                              [boolean]
  -V, --version  show version information                              [boolean]
  -j, --json     print in JSON format

Missing required argument: f"
`;

exports[`end-to-end integration when no arguments passed prints error and help message 1`] = `
"Usage: Utest.js [option]

Options:
  -f, --file     Load a file                                          [required]
  -h, --help     Show help                                             [boolean]
  -v, --version  show version information                              [boolean]
  -V, --version  show version information                              [boolean]
  -j, --json     print in JSON format

Missing required argument: f"
`;

exports[`end-to-end integration when argument -j -f without file is specified should print help message 1`] = `
"Usage: Utest.js [option]

Options:
  -f, --file     Load a file                                          [required]
  -h, --help     Show help                                             [boolean]
  -v, --version  show version information                              [boolean]
  -V, --version  show version information                              [boolean]
  -j, --json     print in JSON format

Missing required argument: f"
`;

exports[`end-to-end integration when argument --json -f without file is specified should print help message 1`] = `
"Usage: Utest.js [option]

Options:
  -f, --file     Load a file                                          [required]
  -h, --help     Show help                                             [boolean]
  -v, --version  show version information                              [boolean]
  -V, --version  show version information                              [boolean]
  -j, --json     print in JSON format

Missing required argument: f"
`;

exports[`end-to-end integration when argument -j --file without file is specified should print help message 1`] = `
"Usage: Utest.js [option]

Options:
  -f, --file     Load a file                                          [required]
  -h, --help     Show help                                             [boolean]
  -v, --version  show version information                              [boolean]
  -V, --version  show version information                              [boolean]
  -j, --json     print in JSON format

Missing required argument: f"
`;

exports[`end-to-end integration when argument --json --file without file is specified should print help message 1`] = `
"Usage: Utest.js [option]

Options:
  -f, --file     Load a file                                          [required]
  -h, --help     Show help                                             [boolean]
  -v, --version  show version information                              [boolean]
  -V, --version  show version information                              [boolean]
  -j, --json     print in JSON format

Missing required argument: f"
`;
